lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1639,11 +1644,16 @@ def obj_has_parent(self, obj, parent):
             if current_parent is None:
                 return False
 
-    def select_resource_pool_by_name(self, resource_pool_name):
-        resource_pool = self.cache.find_obj(self.content, [vim.ResourcePool], resource_pool_name)
-        if resource_pool is None:
+    def select_resource_pool_by_name(self, resource_pool_name, cluster=None):
+        resource_pools = self.cache.find_obj(self.content, [vim.ResourcePool], resource_pool_name, return_list=True)
```

Comment:
self.cache contains function `get_all_objs` which already does this, so we can reuse it directly rather than modifying `find_obj`